### ISSUE TITLE
Upgrade from C124 -> C125 on Win x64 via 1.67.x - Nightly (Checking C125)

### ISSUE THREAD

**Issue by LaurenWags** [brave/brave-browser#38038]
As per process, QA runs through the following cases to ensure that the major chromium bump that's about to get merged into `master` doesn't regress the `Nightly` channel. Once the major chromium bump is merged into `master`, QA will run through a full manual pass on `Windows` & `Android`.

### Startup & Components

- [x] Verify that Brave is only contacting `*.brave.com` endpoints on first launch using either `Charles Proxy`, `Fiddler`, `Wireshark` or `LittleSnitch` (or a similar application)
  - [x] Verify that opening a NTP doesn't trigger any outbound connections related to widgets without user interaction
- [x] Remove the following component folders and ensure that they're being re-downloaded after restarting the browser:
  - [x] `afalakplffnnnlkncjhbmahjfjhmlkal`: `AutoplayWhitelist.dat`, `ExtensionWhitelist.dat`, `ReferrerWhitelist.json` and `Greaselion.json`
  - [x] `CertificateRevocation`
  - [x] `gccbbckogglekeggclmmekihdgdpdgoe`: (Sponsored New Tab Images)
  - [x] `gkboaolpopklhgplhaaiboijnklogmbc`: Brave Ad Block List Catalog
  - [x] `iodkpdagapdfkphljnddpjlldadblomo`: Brave Ad Block Updater
  - [x] `mfddibmblmbccpadfndgakiopmmhebop`: Brave Ad Block List Catalog
  - [x] `oofiananboodjbbmdelgdommihjbkfag`: HTTPSE
  - [x] `Safe Browsing`
- [x] Restart the browser, load `brave://components`, wait for 8 mins and verify that no component shows any errors

**Note:** Always double check `brave://components` to make sure there's no errors/missing version numbers

### Upgrade

- [x] Make sure that data from the last version appears in the new version OK
- [x] Ensure that `brave://version` lists the expected Brave & Chromium versions
- [x] With data from the last version, verify that:
  - [x] Bookmarks on the bookmark toolbar and bookmark folders can be opened
  - [x] Cookies are preserved
  - [x] Installed extensions are retained and work correctly
  - [x] Opened tabs can be reloaded
  - [x] Stored passwords are preserved
  - [x] Sync chain created in previous version is retained
  - [x] Social media-blocking buttons changes are retained
  - [x] Custom filters under brave://settings/shields/filters are retained
  - [x] Custom lists under brave://settings/shields/filters are retained
  - [x] Rewards
    - [x] BAT balance is retained
    - [x] Auto-contribute list is retained
    - [x] Both Tips and Monthly Contributions are retained
    - [x] Panel transactions list is retained
    - [x] Changes to rewards settings are retained
    - [x] Ensure that Auto Contribute is not being enabled when upgrading to a new version if AC was disabled
  - [x] Ads
    - [x] Both `Estimated pending rewards` & `Ad notifications received this month` are retained
    - [x] Changes to ads settings are retained
    - [x] Ensure that ads are not being enabled when upgrading to a new version if they were disabled
    - [x] Ensure that ads are not disabled when upgrading to a new version if they were enabled

**Comment by MadhaviSeelam** [comment]
Verification `In progress` using
```
Brave | 1.67.59 Chromium: 125.0.6422.21 (Official Build) nightly (64-bit)
-- | --
Revision | 7f71207785cb6068bf93b4d1370a67e9c68de72b
OS | Windows 11 Version 23H2 (Build 22631.3527)
```

<details>
<summary>C124 --> C125 on-demand upgrade check - Pending</summary>

1. installed latest nightly `1.67.64 Chromium: 124.0.6367.118 (Official Build) nightly (64-bit) ` and closed Brave
2. created `HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\BraveSoftware\UpdateDev\url` and set `url` as the following:
  * `https://updates.brave.com/service/update2`
4. set `HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\BraveSoftware\Update\ClientState\{C6CB981E-DB30-4876-8639-109F8933582C}\ap` as `nightly` (usually `cr-bump-update-tests-x64` is used but this time around `nightly` was used)
5. relaunched Brave and visited `brave://settings/help` and ensured that new update was being pulled/downloaded
6. ensured that clicking `Relaunch` restarted the browser without any issues
7. ensured Brave was updated to `1.67.59 Chromium: 125.0.6422.21 (Official Build) nightly (64-bit)` without any issues and you can restart the browser

**`step 1`** | **`step 2`** | **`step 3`** | **`step 4`** | **`step 5`**
----------------|----------------|----------------|---------------|------------


</details>

<details>
<summary>C124 --> C125 background upgrade check - Pending</summary>

1. installed latest nightly `1.67.64 Chromium: 124.0.6367.118 (Official Build) nightly (64-bit)` and closed Brave
2. created `HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\BraveSoftware\UpdateDev\url` and set `url` as the following:
  * `https://updates.brave.com/service/update2`
3. set `HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\BraveSoftware\Update\ClientState\{C6CB981E-DB30-4876-8639-109F8933582C}\ap` as `nightly` (usually `cr-bump-update-tests-x64` is used but this time around `nightly` was used)
4. started `BraveSoftwareUpdateTaskMachineUA{3841620D-F169-4927-BDD8-B34AB5FF24F1}` via `Task Scheduler`
5. waited ~30 min or so before launching Brave and ensured it was updated to `1.67.59 Chromium: 125.0.6422.21 (Official Build) nightly (64-bit)`  
6. restarted Brave several times and ensured that there's no issues re-launching Brave once it was updated via the `background`

**`step 1`** | **`step 2`** | **`step 3`** | **`step 4`** | **`step 5`**
----------------|----------------|---------------|----------------|-----------


</details>

<details>
<summary>C125 --> C125 on-demand upgrade check - Pending</summary>

* installed `1.67.59 Chromium: 125.0.6422.21` via `[BraveBrowserNightlySetup_125_1_67_59.exe]
* created `HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\BraveSoftware\UpdateDev\url` and set `url` as the following:
  * `https://updates.brave.com/service/update2`
* set `HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\BraveSoftware\Update\ClientState\{C6CB981E-DB30-4876-8639-109F8933582C}\ap` as `nightly` (usually `cr-bump-update-tests-x64` is used but this time around `nightly` was used)
* relaunched Brave and visited `brave://settings/help` and ensured that new update was being pulled/downloaded
* ensured that clicking `Relaunch` restarted the browser without any issues
* ensured Brave was updated to `1.67.64 Chromium: 125.0.6422.21 (Official Build) nightly (64-bit)`  without any issues and you can restart the browser

**`Example`** | **`Example`** | **`Example`** | **`Example`** | **`Example`**
----------------|----------------|----------------|---------------|----------


</details>
<details>
<summary>C125 --> C125 background upgrade check - Pending</summary>

* installed `1.67.59 Chromium: 125.0.6422.21` via `[BraveBrowserNightlySetup_125_1_67_59.exe]
* created `HKEY_LOCAL_MACHINE\SOFTWARE\BraveSoftware\UpdateDev\url` and set `url` as the following:
  * `https://updates.brave.com/service/update2`
* set `Computer\HKEY_LOCAL_MACHINE\SOFTWARE\BraveSoftware\Update\ClientState\{C6CB981E-DB30-4876-8639-109F8933582C}\ap` as `nightly` (usually `cr-bump-update-tests-x86` is used but this time around `nightly` was used)
* started `BraveSoftwareUpdateTaskMachineUA{3841620D-F169-4927-BDD8-B34AB5FF24F1}` via `Task Scheduler`
* waited ~30min or so before launching Brave and ensured it was updated to `1.67.64 Chromium: 125.0.6422.21 (Official Build) nightly (64-bit)`
* restarted Brave several times and ensured that there's no issues re-launching Brave once it was updated via the `background

**`step 1`** | **`step 2`** | **`step 3`** | **`Example`** | **`Example`**
----------------|----------------|----------------|----------------|------------


</details>

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 10 x86` using the following build(s):

```
Brave | 1.67.64 Chromium: 125.0.6422.26 (Official Build) nightly (32-bit)
-- | --
Revision | bfd2636e5cf7bb0b023414637e9a803310544b77
OS | Windows 10 Version 22H2 (Build 19045.4355)
```

<details>
<summary>C124 --> C125 on-demand upgrade check - PASSED</summary>

* installed `1.67.56 Chromium: 124.0.6367.91` and closed Brave
* created `HKEY_LOCAL_MACHINE\SOFTWARE\BraveSoftware\UpdateDev\url` and set `url` as the following:
  * `https://updates.brave.com/service/update2`
* set `Computer\HKEY_LOCAL_MACHINE\SOFTWARE\BraveSoftware\Update\ClientState\{C6CB981E-DB30-4876-8639-109F8933582C}\ap` as `nightly` (usually `cr-bump-update-tests-x86` is used but this time around `nightly` was used)
* realunched Brave and visited `brave://settings/help` and ensured that new update was being pulled/downloaded
* ensured that clicking `Relaunch` restarted the browser without any issues
* ensured Brave was updated to `1.67.64 Chromium: 125.0.6422.26` without any issues and you can restart the browser

**`Example`** | **`Example`** | **`Example`** | **`Example`** 
----------------|----------------|----------------|---------------
![image](https://github.com/brave/brave-browser/assets/2602313/c12a4249-0ab5-490a-a140-f693df6f97c2) | ![image](https://github.com/brave/brave-browser/assets/2602313/13a40d9b-3d9e-4aff-9469-27d7b4aac607) | ![image](https://github.com/brave/brave-browser/assets/2602313/777b4af5-3666-4ad1-911d-8cf1ae68d56c) | ![image](https://github.com/brave/brave-browser/assets/2602313/2b2677b9-1a26-4a81-8395-e81b843c2b41)
</details>

<details>
<summary>C125 --> C125 on-demand upgrade check - PASSED</summary>

* installed `1.67.59 Chromium: 125.0.6422.21` via `BraveBrowserStandaloneNightlySetup32_125_1_67_59.exe` and closed Brave
* created `HKEY_LOCAL_MACHINE\SOFTWARE\BraveSoftware\UpdateDev\url` and set `url` as the following:
  * `https://updates.brave.com/service/update2`
* set `Computer\HKEY_LOCAL_MACHINE\SOFTWARE\BraveSoftware\Update\ClientState\{C6CB981E-DB30-4876-8639-109F8933582C}\ap` as `nightly` (usually `cr-bump-update-tests-x86` is used but this time around `nightly` was used)
* realunched Brave and visited `brave://settings/help` and ensured that new update was being pulled/downloaded
* ensured that clicking `Relaunch` restarted the browser without any issues
* ensured Brave was updated to `1.67.64 Chromium: 125.0.6422.26` without any issues and you can restart the browser

**`Example`** | **`Example`** | **`Example`** | **`Example`** 
----------------|----------------|----------------|----------------
![image](https://github.com/brave/brave-browser/assets/2602313/955229a6-c463-4002-9ac2-80edd4d4c269) | ![image](https://github.com/brave/brave-browser/assets/2602313/c8b0559f-be92-4632-a7e9-2301b1568eca) | ![image](https://github.com/brave/brave-browser/assets/2602313/3c02411f-56ab-4399-ae2a-aceeb73fd9f4) | ![image](https://github.com/brave/brave-browser/assets/2602313/d0190bb8-7175-48d7-8f16-fbee33f53223)

`x86` delta logs from `C:\ProgramData\BraveSoftware\Update\Log\BraveUpdate.log`:

```
[05/03/24 15:52:30.527][BraveUpdate:goopdate][5604:5500][Installing][display name: Brave Nightly][app id: {C6CB981E-DB30-4876-8639-109F8933582C}][installer path: C:\Program Files\BraveSoftware\Update\Install\{5003DD2A-8063-472D-9FCF-494833213643}\brave_installer-delta-ia32.exe][manifest args: --chrome-sxs][installer data: ][untrusted data: ][priority: 10]
[05/03/24 15:52:30.527][BraveUpdate:goopdate][5604:5500][Running installer][C:\Program Files\BraveSoftware\Update\Install\{5003DD2A-8063-472D-9FCF-494833213643}\brave_installer-delta-ia32.exe][--chrome-sxs][{C6CB981E-DB30-4876-8639-109F8933582C}]
[05/03/24 15:52:49.558][BraveUpdate:goopdate][5604:5500][InstallerResult][{C6CB981E-DB30-4876-8639-109F8933582C}][0]
[05/03/24 15:52:49.558][BraveUpdate:goopdate][5604:5500][InstallApp returned][0x0][{C6CB981E-DB30-4876-8639-109F8933582C}][type:1][code: 30][][]
[05/03/24 15:52:49.569][BraveUpdate:goopdate][5604:5500][Started process][11480]
[05/03/24 15:52:49.577][BraveUpdate:goopdate][5604:5500][LogEventHelper][4][12][1][[
Application update/install.
pid=5604, ver=1.3.361.149, machine=1, extern=1, private
App={C6CB981E-DB30-4876-8639-109F8933582C}, Ver=125.1.67.59, Download result=0x00000000
App={C6CB981E-DB30-4876-8639-109F8933582C}, Ver=125.1.67.59, Install result=0x00000000
]][]
```

</details>

<details>
<summary>C124 --> C125 background upgrade check - PASSED</summary>

* installed `1.67.56 Chromium: 124.0.6367.91` and closed Brave
* created `HKEY_LOCAL_MACHINE\SOFTWARE\BraveSoftware\UpdateDev\url` and set `url` as the following:
  * `https://updates.brave.com/service/update2`
* set `Computer\HKEY_LOCAL_MACHINE\SOFTWARE\BraveSoftware\Update\ClientState\{C6CB981E-DB30-4876-8639-109F8933582C}\ap` as `nightly` (usually `cr-bump-update-tests-x86` is used but this time around `nightly` was used)
* started `BraveSoftwareUpdateTaskMachineUA{842DF633-33C5-4967-A2D4-A112EC962E62}` via `Task Scheduler`
* waited ~30min or so before launching Brave and ensured it was updated to `1.67.64 Chromium: 125.0.6422.26`
* restarted Brave several times and ensured that there's no issues re-launching Brave once it was updated via the `background`

**`Example`** | **`Example`** | **`Example`** | **`Example`**
----------------|----------------|---------------|-----------------
![image](https://github.com/brave/brave-browser/assets/2602313/c0014cae-e976-428b-8823-a99c17f805f8) | ![image](https://github.com/brave/brave-browser/assets/2602313/51d2bd15-3c2b-49bf-8e79-654077cc2645) | ![image](https://github.com/brave/brave-browser/assets/2602313/1f194c9b-ac83-40ad-9fde-af1158e21796) | ![image](https://github.com/brave/brave-browser/assets/2602313/1838b1bd-fecf-4ec6-9808-e42452f17600)

</details>

<details>
<summary>C125 --> C125 background upgrade check - PASSED</summary>

* installed `1.67.59 Chromium: 125.0.6422.21` via `BraveBrowserStandaloneNightlySetup32_125_1_67_59.exe` and closed Brave
* created `HKEY_LOCAL_MACHINE\SOFTWARE\BraveSoftware\UpdateDev\url` and set `url` as the following:
  * `https://updates.brave.com/service/update2`
* set `Computer\HKEY_LOCAL_MACHINE\SOFTWARE\BraveSoftware\Update\ClientState\{C6CB981E-DB30-4876-8639-109F8933582C}\ap` as `nightly` (usually `cr-bump-update-tests-x86` is used but this time around `nightly` was used)
* started `BraveSoftwareUpdateTaskMachineUA{EB87DF49-99D6-44D9-BDA3-7736229A56EE}` via `Task Scheduler`
* waited ~30min or so before launching Brave and ensured it was updated to `1.67.64 Chromium: 125.0.6422.26`
* restarted Brave several times and ensured that there's no issues re-launching Brave once it was updated via the `background

**`Example`** | **`Example`** | **`Example`** | **`Example`**
----------------|----------------|----------------|----------------
![image](https://github.com/brave/brave-browser/assets/2602313/77f3a596-34d2-4b69-af42-412dad715d7c) | ![image](https://github.com/brave/brave-browser/assets/2602313/d54c1976-d346-4875-9b03-46e3d687fc28) | ![image](https://github.com/brave/brave-browser/assets/2602313/02956286-bdda-491c-96d6-aa55af3519e9) | ![image](https://github.com/brave/brave-browser/assets/2602313/94ce3a61-9e4b-46bd-8b69-c2076f0025e8)

`x86` delta logs from `C:\ProgramData\BraveSoftware\Update\Log\BraveUpdate.log`:

```
[05/03/24 16:29:10.177][BraveUpdate:goopdate][7624:5540][Installing][display name: Brave Nightly][app id: {C6CB981E-DB30-4876-8639-109F8933582C}][installer path: C:\Program Files\BraveSoftware\Update\Install\{D2933995-079D-4153-A33C-8A0D691FBCC8}\brave_installer-delta-ia32.exe][manifest args: --chrome-sxs][installer data: ][untrusted data: ][priority: 0]
[05/03/24 16:29:10.177][BraveUpdate:goopdate][7624:5540][Running installer][C:\Program Files\BraveSoftware\Update\Install\{D2933995-079D-4153-A33C-8A0D691FBCC8}\brave_installer-delta-ia32.exe][--chrome-sxs][{C6CB981E-DB30-4876-8639-109F8933582C}]
[05/03/24 16:29:30.302][BraveUpdate:goopdate][7624:5540][InstallerResult][{C6CB981E-DB30-4876-8639-109F8933582C}][0]
[05/03/24 16:29:30.302][BraveUpdate:goopdate][7624:5540][InstallApp returned][0x0][{C6CB981E-DB30-4876-8639-109F8933582C}][type:1][code: 2][]["C:\Program Files\BraveSoftware\Brave-Browser-Nightly\Application\brave.exe" --from-installer]
[05/03/24 16:29:30.302][BraveUpdate:goopdate][7624:5540][Started process][9436]
[05/03/24 16:29:30.302][BraveUpdate:goopdate][7624:5540][LogEventHelper][4][12][1][[
Application update/install.
pid=7624, ver=1.3.361.149, machine=1, extern=1, private
App={C6CB981E-DB30-4876-8639-109F8933582C}, Ver=125.1.67.59, Download result=0x00000000
App={C6CB981E-DB30-4876-8639-109F8933582C}, Ver=125.1.67.59, Install result=0x00000000
]][]
```

</details>
